File: sorting/selection_sort.py
"""
Iterates over the list and swapping each 
element with the minimum (or maximum) element 
found in the unsorted list with that in the sorted list.
"""


def selection_sort(lst):
    for i in range(len(lst)):
        min_index = i
        for j in range(i + 1, len(lst)):
            if lst[min_index] > lst[j]:
                min_index = j
        lst[i], lst[min_index] = lst[min_index], lst[i]

File: sorting/selectionSort.cpp
#include <vector>

/*
Iterates over the array and swapping each
element with the minimum (or maximum) element
found in the unsorted array with that in the sorted array.
*/

void selectionSort(std::vector<int> arr)
{
    for (int i = 0; i < arr.size(); i++)
    {
        int minIndex = i;
        for (int j = i + 1; j < arr.size(); j++)
        {
            if (arr[minIndex] > arr[j])
                minIndex = j;
        }
        arr[i], arr[minIndex] = arr[minIndex], arr[i];
    }
}


